gcloud/storage/bucket.py

```diff
@@ -205,6 +205,11 @@ def path(self):
 
         return self.path_helper(self.name)
 
+    @property
```

Comment:
I've got a sense that this property (maybe the others too) should be moved up to before the methods which use it (at least, I was concerned about the `_client` vs. `client` usage in this diff).